source/adapters/cuda/device.cpp

```diff
@@ -1105,6 +1104,13 @@ UR_APIEXPORT ur_result_t UR_APICALL urDeviceGetInfo(ur_device_handle_t hDevice,
                              CU_DEVICE_ATTRIBUTE_COMPUTE_CAPABILITY_MAJOR) >= 9;
     return ReturnValue(static_cast<bool>(Value));
   }
+  case UR_DEVICE_INFO_USE_NATIVE_ASSERT:
+    // CUDA doesn't actually support asserts natively so this results in
```

Comment:
What does native assert mean if cuda doesn't support it? cuda has an implementation for `std::assert` etc. It has a sycl test here https://github.com/intel/llvm/blob/sycl/sycl/test-e2e/Assert/assert_in_kernels.hpp
Hip also supports it in theory, but has some issues on some old (but officially supported) devices https://github.com/ROCm/HIP/issues/3368

Reply:
Nice, will delete. I was basing my assumptions on the discussion I found on https://github.com/intel/llvm/pull/4604 when I was trying to figure out why we were reporting support for that cl extension